source/gamemodes/BoostMod.cpp

```diff
@@ -0,0 +1,138 @@
+
+
+/// <summary>Checks and updates boost every game tick.</summary>
+/// <remarks>Gets called on 'Function GameEvent_Soccar_TA.Active.Tick'.</remarks>
+/// <param name="server"><see cref="ServerWrapper"/> instance of the current server</param>
+/// <param name="params">Delay since last update</param>
+void BoostMod::onTick(ServerWrapper server, void*)
+{
+	if (server.IsNull()) {
+		return;
+	}
+
+	float maxBoost = 1.0f;
+
+	if (generalBoostSettings.enabled) {
+		maxBoost = (float)generalBoostSettings.maxBoost / 100.0f;
+	}
+
+	auto teams = server.GetTeams();
+
+	for (int teamIdx = 0; teamIdx < teams.Count(); teamIdx++) {
+		float currentTeamMax = maxBoost;
+
+		if (teamIdx < teamsBoostSettings.size() && teamsBoostSettings[teamIdx].enabled) {
+			currentTeamMax = (float)teamsBoostSettings[teamIdx].maxBoost / 100.0f;
+		}
+		
+		if (generalBoostSettings.enabled || teamsBoostSettings[teamIdx].enabled) {
+			auto team = teams.Get(teamIdx);
+			if (team.IsNull()) { continue; }
+			auto players = team.GetMembers();
+
+			for (int playerIdx = 0; playerIdx < players.Count(); playerIdx++) {
+				auto player = players.Get(playerIdx);
+				if (player.IsNull()) { continue; }
+
+				auto car = player.GetCar();
+				if (car.IsNull()) { continue; }
+
+				auto boost = car.GetBoostComponent();
+				if (boost.IsNull()) { continue; }
+
+				if (boost.GetCurrentBoostAmount() > currentTeamMax) {
+					boost.SetBoostAmount(currentTeamMax);
+				}
+			}
+		}
+	}
+}
+
+/// Util function to render a single set of boost related options
+void BoostMod::RenderSingleOption(BoostModValues& boostSettings)
+{
+	ImGui::Spacing();
+	if (ImGui::SwitchCheckbox((" Enable " + boostSettings.displaySuffix).c_str(), 
+		boostSettings.enabled)) 
+	{
+		boostSettings.enabled = !boostSettings.enabled;
+	}
+	if (boostSettings.enabled) {
+		//TODO: implement boost amount type modifier.
+		//ImGui::SliderArray(("Boost Amount Type" + boostSettings.displaySuffix).c_str(), &boostSettings.boostAmountType, boostAmounts);
+		ImGui::SliderInt(("Boost Limit " + boostSettings.displaySuffix).c_str(),
+			&boostSettings.maxBoost, 0, 100, "%d%%");
+	}
+	else {
+		ImGui::BeginDisabled();
+		//ImGui::SliderArray(("Boost Amount Type" + boostSettings.displaySuffix).c_str(), &boostSettings.boostAmountType, boostAmounts);
+		ImGui::SliderInt(("Boost Limit " + boostSettings.displaySuffix).c_str(), &boostSettings.maxBoost, 0, 100, "%d%%");
+		ImGui::EndDisabled();
+	}
+	ImGui::Spacing();
+}
+
+/// <summary>Renders the available options for the gamemode.</summary>
+void BoostMod::RenderOptions()
+{
+	// general modifier (everyone)
+	ImGui::Text("General:");
+	ImGui::Indent(20);
+	RenderSingleOption(generalBoostSettings);
+	ImGui::Unindent();
+
+	
+	// team specific
+	if (ImGui::CollapsingHeader("Team modifiers:")) {
+		ImGui::Indent(20);
+		for (int teamIdx = 0; teamIdx < teamsBoostSettings.size(); teamIdx++) {
+			if (ImGui::CollapsingHeader(teamsBoostSettings[teamIdx].displaySuffix.c_str())) {
```

Comment:
Instead of hardcoding the team names to avoid having multiple ImGui widgets with the same label, you can use `PushID(int_id)`,
```suggestion
		for (int teamIdx = 0; teamIdx < teamsBoostSettings.size(); teamIdx++) {
			ImGui::PushID(int teamIdx);
			if (ImGui::CollapsingHeader(teamsBoostSettings[teamIdx].displaySuffix.c_str())) {
```
and the get the actual team name with `TeamInfoWrapper::GetTeamName()`

Reply:
GetTeamName() seems to be returning "Team" for both teams in normal local match so I ended up hardcoding the names. I'm reading about pushID popId now.